### DISCUSSION TITLE
Prices of Commodity purchases

### DISCUSSION THREAD

**Discussion by HackHerz** [ananthakumaran/paisa#203] [Q&A] [answered]
I think i am modelling ETF purchases like i am supposed to:

```
2024-03-25 * Investment
    Assets:Checking
    Assets:Equity:ETF                       0.04 "SP500" @@ €20
```

I am using `@@` because my broker is told to use 20€ to buy ETF shares, if i specify the unit price i get rounding mistakes and not exactly 20€ are calculated in ledger.

However, even if i specify the unit price Paisa does not seem to take that into account? If i have a look at the gain page it shows my investments as "0.04", even though i have specified the commodity prices in paisa (they are also correctly fetched from yahoo and displayed in ledger > price). I have also tried with the examples of the demo instance: does not seem to work on my instance. I am using the latest docker image.

<img width="464" alt="Bildschirmfoto 2024-03-28 um 16 33 01" src="https://github.com/ananthakumaran/paisa/assets/2885148/74bf0a1e-501e-4751-af77-498773e0f092">

Am i missing something? I am also getting a warning from the "doctor" that unit prices in that purchase do not match with the fetched prices (i think yahoo only displays the price after market closure for a given day, so makes sense).

**Comment by TMaddox** [comment]
This works for me, I haven't tried it but your issue might be the "" or the € ahead of the amount. FYI there are good examples at: https://ledger-cli.org/doc/ledger3.html

2020/01/01 ETF
    Assets:Exchanges:Flatex:ETF_NAME                  10 ETF @ 100 EUR
    Assets:Exchanges:Flatex:EUR             -1010 EUR
    Expenses:Investing:Fees:Flatex             10 EUR

**Reply by HackHerz** [reply]
Well, yes and no. Your comment gave me the right direction.

So for ledger itself it makes no difference if i write `EUR 100`  or `100 EUR`, the output is the same. But the issue is, that i am using the € symbol. `EUR 100` results in paisa parsing the output correctly, `€ 100` is not parsed correctly. So i think it is a issue with paisa and not with ledger or my modelling.

**Reply by HackHerz** [reply, marked as answer]
Actually that is not true, in my configuration the default currency was set to "EUR" and not to "€", now it is working. My bad. Thanks for helping me track this down.
